recipes/primesieve/all/conanfile.py

```diff
@@ -0,0 +1,103 @@
+
+class PackageConan(ConanFile):
+    name = "primesieve"
+    description = "Fast prime number generator"
+    license = "BSD-2-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/kimwalisch/primesieve"
+    topics = ("math", "prime-numbers", "sieve-of-eratosthenes")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_multiarch": [True, False],
+        "with_msvc_crt_static": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_multiarch": True,
+        "with_msvc_crt_static": False,
+    }
+    package_type = "library"
+
+    @property
+    def _min_cppstd(self):
+        return 11
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        
+        if self.settings.compiler == "gcc" and Version(self.settings.compiler.version) <= "5":
+            raise ConanInvalidConfiguration("GCC<=5 is currently not supported. Contributions with fixes are welcome.")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["BUILD_PRIMESIEVE"] = False
+        tc.variables["BUILD_DOC"] = False
+        tc.variables["BUILD_EXAMPLES"] = False
+        tc.variables["BUILD_TESTS"] = False
+        tc.variables["WITH_MULTIARCH"] = self.options.with_multiarch
+        tc.variables["BUILD_STATIC_LIBS"] = not self.options.shared
+        if is_msvc(self):
+            tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0091"] = "NEW"
+            tc.variables["WITH_MSVC_CRT_STATIC"] = self.options.with_msvc_crt_static
+            tc.variables["USE_MSVC_RUNTIME_LIBRARY_DLL"] = not is_msvc_static_runtime(self)
+        tc.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, pattern="COPYING", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
+        cmake.install()
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rm(self, "*.la", os.path.join(self.package_folder, "lib"))
+        rm(self, "*.pdb", os.path.join(self.package_folder, "lib"))
+        rm(self, "*.pdb", os.path.join(self.package_folder, "bin"))
+        if is_msvc(self) and self.options.shared:
```

Comment:
Any insight into this rename? It looks funny that it gets generated with this name in the first place :)

Reply:
It seems this is set to avoid issues when packaging both static and shared libraries in the same package. This doesn't apply to us, so I think it's fine to do this? Unless we have a way of telling Conan that this is how it's done.

https://github.com/kimwalisch/primesieve/blob/master/CMakeLists.txt#L138

Reply:
Please, keep the original name, it should work as usual, right? I remember having other projects having this same prefix when exporting dll libs

Reply:
It would fail without this change, I presume it tries to look for "primesieve".lib but it's "primesieve.dll".lib. I'm fine with changing it back, I just don't know how I would get it to work.

Reply:
@uilianries @RubenRBS Many more recipes are doing this rename. You can run a grep and see this ad-hoc helper function `fix_msvc_libname`. The problem is coming from the hook error:
```
primesieve/11.2: WARN: Using the new toolchains and generators without specifying a build profile (e.g: -pr:b=default) is discouraged and might cause failures and unexpected behavior
[HOOK - conan-center.py] post_package_info(): ERROR: [LIBRARY DOES NOT EXIST (KB-H054)] Component primesieve::primesieve library 'primesieve' is listed in the recipe, but not found installed at self.cpp_info.libdirs. Make sure you compiled the library correctly. If so, then the library name should probably be fixed. Otherwise, then the component should be removed. (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H054-LIBRARY-DOES-NOT-EXIST) 
ERROR: 
	ConanException: [HOOK - conan-center.py] post_package_info(): Some checks failed running the hook, check the output
```
@Ahajha We just created an issue to review this process https://github.com/conan-io/hooks/issues/528 Let's figure out if there is a bug in the hooks so we can use the original library name instead of the renamed one.